Travis builds suceed even if cmake fails

START CONFIGURE
==================================================
Using default myconfig.
>cmake -DWITH_PYTHON=OFF  /home/espresso/travis
-- The C compiler identification is GNU 5.4.0
-- The CXX compiler identification is GNU 5.4.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Config file: /home/espresso/travis/src/core/myconfig-default.hpp
-- Performing Test result__PRETTY_FUNCTION__
-- Performing Test result__PRETTY_FUNCTION__ - Success
-- Found PythonInterp: /usr/bin/python (found version "2.7.12") 
-- Found Cython version 0.23
-- Found Cython: /usr/bin/cython (Required is at least version "0.23") 
-- Found NumPy: /usr/lib/python2.7/dist-packages/numpy/core/include (found version "1.11.0") 
-- NumPy ver. 1.11.0 found (include: /usr/lib/python2.7/dist-packages/numpy/core/include)
CMake Error at /usr/share/cmake-3.5/Modules/FindPackageHandleStandardArgs.cmake:148 (message):
  The Python module enum was not found.  (missing: PY_ENUM)
Call Stack (most recent call first):
  /usr/share/cmake-3.5/Modules/FindPackageHandleStandardArgs.cmake:388 (_FPHSA_FAILURE_MESSAGE)
  cmake/FindPythonModule.cmake:23 (find_package_handle_standard_args)
  CMakeLists.txt:147 (find_python_module)


-- Configuring incomplete, errors occurred!
See also "/home/espresso/travis/CMakeFiles/CMakeOutput.log".
maintainer/travis/build_docker.sh: line 51: [: =: unary operator expected


Done. Your build exited with 0.

Iirc this is fixed in some pr

On Mon, Aug 28, 2017, 8:56 PM RudolfWeeber <notifications@github.com> wrote:

> START CONFIGURE
>
> Using default myconfig.
>
> cmake -DWITH_PYTHON=OFF /home/espresso/travis
> -- The C compiler identification is GNU 5.4.0
> -- The CXX compiler identification is GNU 5.4.0
> -- Check for working C compiler: /usr/bin/cc
> -- Check for working C compiler: /usr/bin/cc -- works
> -- Detecting C compiler ABI info
> -- Detecting C compiler ABI info - done
> -- Detecting C compile features
> -- Detecting C compile features - done
> -- Check for working CXX compiler: /usr/bin/c++
> -- Check for working CXX compiler: /usr/bin/c++ -- works
> -- Detecting CXX compiler ABI info
> -- Detecting CXX compiler ABI info - done
> -- Detecting CXX compile features
> -- Detecting CXX compile features - done
> -- Config file: /home/espresso/travis/src/core/myconfig-default.hpp
> -- Performing Test result__PRETTY_FUNCTION__
> -- Performing Test result__PRETTY_FUNCTION__ - Success
> -- Found PythonInterp: /usr/bin/python (found version "2.7.12")
> -- Found Cython version 0.23
> -- Found Cython: /usr/bin/cython (Required is at least version "0.23")
> -- Found NumPy: /usr/lib/python2.7/dist-packages/numpy/core/include (found
> version "1.11.0")
> -- NumPy ver. 1.11.0 found (include:
> /usr/lib/python2.7/dist-packages/numpy/core/include)
> CMake Error at
> /usr/share/cmake-3.5/Modules/FindPackageHandleStandardArgs.cmake:148
> (message):
> The Python module enum was not found. (missing: PY_ENUM)
> Call Stack (most recent call first):
> /usr/share/cmake-3.5/Modules/FindPackageHandleStandardArgs.cmake:388
> (_FPHSA_FAILURE_MESSAGE)
> cmake/FindPythonModule.cmake:23 (find_package_handle_standard_args)
> CMakeLists.txt:147 (find_python_module)
>
> -- Configuring incomplete, errors occurred!
> See also "/home/espresso/travis/CMakeFiles/CMakeOutput.log".
> maintainer/travis/build_docker.sh: line 51: [: =: unary operator expected
>
> Done. Your build exited with 0.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/espressomd/espresso/issues/1358>, or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA2i36QrWUANb_fZ8f56chfbuV7D91pZks5scw1KgaJpZM4PE62Q>
> .
>

--- a/maintainer/travis/build_cmake.sh
+++ b/maintainer/travis/build_cmake.sh
@@ -163,5 +163,5 @@ if $with_coverage; then
     lcov --remove coverage.info '/usr/*' --output-file coverage.info # filter out system
     lcov --list coverage.info #debug info
     # Uploading report to CodeCov
-    bash <(curl -s https://codecov.io/bash) || echo "Codecov did not collect coverage reports"
+    bash <(curl -s https://codecov.io/bash) -t d1e004ab-1d37-4bde-af79-549b2a20ad25 || echo "Codecov did not collect coverage reports"
 fi

--- a/maintainer/travis/build_cmake.sh
+++ b/maintainer/travis/build_cmake.sh
@@ -7,7 +7,6 @@
 # notice and this notice are preserved.  This file is offered as-is,
 # without any warranty.
 
-
 # HELPER FUNCTIONS
 
 # output value of env variables
@@ -121,7 +120,7 @@ else
     cp $myconfig_file $builddir/myconfig.hpp
 fi
 
-cmd "cmake $cmake_params $srcdir" || exit $?
+cmd "cmake $cmake_params $srcdir" || exit 1
 end "CONFIGURE"
 
 # BUILD
@@ -163,5 +162,5 @@ if $with_coverage; then
     lcov --remove coverage.info '/usr/*' --output-file coverage.info # filter out system
     lcov --list coverage.info #debug info
     # Uploading report to CodeCov
-    bash <(curl -s https://codecov.io/bash) -t d1e004ab-1d37-4bde-af79-549b2a20ad25 || echo "Codecov did not collect coverage reports"
+    bash <(curl -s https://codecov.io/bash) || echo "Codecov did not collect coverage reports"
 fi

--- a/maintainer/travis/build_docker.sh
+++ b/maintainer/travis/build_docker.sh
@@ -2,6 +2,11 @@
 
 ENV_FILE=$(mktemp esXXXXXXX.env)
 
+ci_env=""
+if [[ ! -z ${with_coverage+x} ]]; then 
+  ci_env=`bash <(curl -s https://codecov.io/env)`
+fi
+
 cat > $ENV_FILE <<EOF
 insource=$insource
 cmake_params=$cmake_params
@@ -19,7 +24,7 @@ fi
 
 if [ "$TRAVIS_OS_NAME" = "linux" ]; then
 	image=espressomd/espresso-$image:latest
-	docker run -u espresso --env-file $ENV_FILE -v ${PWD}:/travis -it $image /bin/bash -c "cp -r /travis .; cd travis && maintainer/travis/build_cmake.sh" || exit 1
+	docker run $ci_env -u espresso --env-file $ENV_FILE -v ${PWD}:/travis -it $image /bin/bash -c "cp -r /travis .; cd travis && maintainer/travis/build_cmake.sh" || exit 1
 elif [ "$TRAVIS_OS_NAME" = "osx" ]; then
 	brew install cmake || brew upgrade cmake
 	case "$image" in
@@ -46,9 +51,4 @@ elif [ "$TRAVIS_OS_NAME" = "osx" ]; then
 
 	export TMPDIR=/tmp
 	maintainer/travis/build_cmake.sh || exit 1
-fi
-
-if [ "$with_coverage" = "true" ]; then
-    cd ${PWD}
-    curl -s https://codecov.io/bash | bash -
 fi

